### PR TITLE
Add support for rustfmt

### LINKED ISSUE #91: Use rustfmt for code formatting
The old format button used `rustc --pretty`.

### PR DESCRIPTION
**Pull request by jonas-schievink**
Closes #91

### CODE REVIEW

**Comment by jonas-schievink** [comment]
Parse errors when running rustfmt are currently displayed without color. Note sure if that's okay. I suspect I could set `TERM=xterm` and see if it fixes it. Might also be related to #174

File: init.sh [hunk 1]
```diff
@@ -39,5 +39,6 @@ for channel in stable beta nightly; do
 	[[ -d root-$channel ]] && mv root-$channel root-${channel}.old
 	mv root-${channel}.new root-$channel
 	[[ -d root-${channel}.old ]] && rm -rf root-${channel}.old
+	arch-chroot root-$channel cargo install --root /usr --vers 0.3 rustfmt
```
Comment (alexcrichton) [inline init.sh]: I wonder if this should drop the `--vers`? Is there any reason to *not* track whatever master is for `rustfmt`?

**Comment by alexcrichton** [comment]
Nice!

File: init.sh (hunk 1 above)
Comment (jonas-schievink) [inline init.sh]: Technically, `rustfmt` is free to break its interface in 0.4, so I did it to make sure that doesn't happen

Comment (alexcrichton) [inline init.sh]: Eh this is using quite a small surface area of rustfmt so it seems like the breakage window is unlikely? We could always update manually if anything goes awry

Comment (jonas-schievink) [inline init.sh]: Sure, I just did this to be extra safe.

**Pull request merged**